src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -532,5 +532,85 @@ private static ILoggerFactory ConfigureLoggerInstance(LogLevel loglevel)
                     .SetMinimumLevel(loglevel);
             });
         }
+
+        internal static async Task ShowOpenApiDocument(string openapi, FileInfo output, LogLevel logLevel, CancellationToken cancellationToken)
+        {
+            using var loggerFactory = Logger.ConfigureLogger(logLevel);
+            var logger = loggerFactory.CreateLogger<OpenApiService>();
+            try
+            {
+                if (string.IsNullOrEmpty(openapi))
+                {
+                    throw new ArgumentNullException(nameof(openapi));
+                }
+                var stream = await GetStream(openapi, logger, cancellationToken);
+
+                OpenApiDocument document;
+                Stopwatch stopwatch = Stopwatch.StartNew();
+                using (logger.BeginScope($"Parsing OpenAPI: {openapi}", openapi))
+                {
+                    stopwatch.Start();
+
+                    var result = await new OpenApiStreamReader(new OpenApiReaderSettings
+                    {
+                        RuleSet = ValidationRuleSet.GetDefaultRuleSet()
+                    }
+                    ).ReadAsync(stream);
+
+                    logger.LogTrace("{timestamp}ms: Completed parsing.", stopwatch.ElapsedMilliseconds);
+
+                    document = result.OpenApiDocument;
+                    var context = result.OpenApiDiagnostic;
+                    if (context.Errors.Count != 0)
+                    {
+                        using (logger.BeginScope("Detected errors"))
+                        {
+                            foreach (var error in context.Errors)
+                            {
+                                logger.LogError(error.ToString());
+                            }
+                        }
+                    }
+                    stopwatch.Stop();
+                }
+
+                using (logger.BeginScope("Creating diagram"))
+                {
+                    // Create OpenApiUrlTree from document
+
+                    using var file = new FileStream(output.FullName, FileMode.Create);
+                    var writer = new StreamWriter(file);
```

Comment:
I did `using` on the FileStream.  Why would I do using on StreamWriter as it doesn't own any unmanaged resources?  That was sort of a rhetorical question. I hate that we have extended the use of Dispose to "also sometime do some other stuff".  All StreamWriter does in its dispose is call Flush. The only reason it implements IDisposable is because TextWriter is disposable but doesn't actually have any code at all in it's Dispose method.  I will accept this change and remove the redundant flush call.  I just felt the need to register my objection to this pattern.